### PR TITLE
Handheld arcade gaming from Josh and hidden items

### PR DESCRIPTION
**Pull request by penny64**
[FEATURE] [QOL]
## About the PR 

This PR introduces two new devices, a handheld Robustris Pro

image

And a handheld arcade- the randomly generated fighting one

image

It also creates a new capsule- "game capsule" which may contain either game, which is hidden in the list of hacked products on the Capsule Vendor.

image

image

Robustris Pro is additionally on offer from Josh- only Robustris Pro as Josh only sells good games.
This will give players two ways to get ahold of these, not to mention that they are scannable so players may mass produce them.

image

image

## Why's this needed? 

These were created as part of the greater motivation to add hidden items to all station-level-vending-machines. It's satisfying for players to decide to try to hack a new machine, and to find that it does in fact have secret inventory.
It also ends up that this is a fun and cool concept that people like, which is why Josh offers them- they are definitely radical enough for josh. 

Finally, they are very cute

https://user-images.githubusercontent.com/13049028/115230858-c04bdd00-a0c9-11eb-8536-434444112532.mp4

Both with animated sprites by me- based on the Geiger counter :)

## Changelog

```changelog
(u)Penny
(*)Handheld arcade machines are now available from Josh
```

### CODE REVIEW

**Comment by Carbadox** [comment]
Could there alternatively be additonal ways to consistently get Robustris? Perhaps a Spacebucks purchase or different vendor? More people would be able to make use of this if it was available outside of just a merchant put behind Quartermaster access.

**Comment by penny64** [comment]
> 
> 
> Could there alternatively be additonal ways to consistently get Robustris? Perhaps a Spacebucks purchase or different vendor? More people would be able to make use of this if it was available outside of just a merchant put behind Quartermaster access.

While not mentioned in the changelog, I do mention these are in the hidden menu of the capsule vendor!

image

**Comment by Carbadox** [comment]
Ah, alright I thought that's what it said the first time I read it and then I got to the end and thought it wasn't the case. Thanks for clarifying

**Comment by penny64** [comment]
@Carbadox tried to make the language more clear :)

File: code/modules/economy/traders/josh.dm [hunk 1]
```diff
@@ -117,7 +118,14 @@
...
 /* Josh sells this stuff */
-
+/datum/commodity/trader/josh/robustris
+	comname = "Sick Gaming Device"
+	comtype = /obj/item/toy/handheld/robustris
+	possible_names = list("You don't have the latest equipment? Let me hook you up bro.",
+	"Looking for some hardcore gaming dude? We got a sweet set.",
+	"Get pumped with some tight games my bro-nado.")
+
+	price_boundary = list(50,300)
```
Comment (Azrun) [inline code/modules/economy/traders/josh.dm:128]:
```suggestion
	price_boundary = list(50,300)
	possible_names = list("You don't have the latest equipment? Let me hook you up bro.",
	"Looking for some hardcore gaming dude? We got a sweet set.",
	"Get pumped with some tight games my bro-nado.")

```

File: code/obj/item/toys.dm [hunk 2]
```diff
@@ -179,3 +180,56 @@
...
+
+
+/obj/item/toy/handheld/robustris
+	icon_state = "arcade-robustris"
+	name = "Robustris Pro"
+
+/obj/item/toy/handheld/arcade
+	arcademode = TRUE
+	icon_state = "arcade-adventure"
+/obj/item/item_box/figure_capsule/gaming_capsule
```
Comment (Azrun) [inline code/obj/item/toys.dm:225]:
```suggestion
	icon_state = "arcade-adventure"
	
/obj/item/item_box/figure_capsule/gaming_capsule
```

**Comment by Azrun** [comment]
Thematically:
How does a gashapon have a secret items that aren't just the contents of the capsules.  I feel like this devalues `/obj/machinery/vending/capsule`, it feels kinda shoehorned.

Mechanically:
Cool!

**Comment by penny64** [comment]
> Thematically:
> How does a gashapon have a secret items that aren't just the contents of the capsules. I feel like this devalues `/obj/machinery/vending/capsule`, it feels kinda shoehorned.
> 
> Mechanically:
> Cool!



> Thematically:
> How does a gashapon have a secret items that aren't just the contents of the capsules. I feel like this devalues `/obj/machinery/vending/capsule`, it feels kinda shoehorned.

@Azrun same with all hidden items- NT employees put it there without asking their boss!

**Comment by Azrun** [comment]
Mechanical behavior of hidden items seems in conflict with the sprite.  There is one reservoir of randomized objects.  
In my head it would work if there was a separate reservoir for capsules to allow for a separate selection.  

The figurine case is a completely separate object that isn't randomized, so that is _fine_.

I understand all these things are abstractions, but right now I see the sprite and immediately know what it is and how it works.  I feel these changes would make that a little less clear. 🤷 

> Also given that the product's role to fill is filling the hidden menu, it was invented backwards, I think it's most thematically important that a hacked capsule vendor just has different capsule!

I guess what I'm getting as I don't think hidden menu items make a lot sense to me as presented.

**Comment by penny64** [comment]
> I guess what I'm getting as I don't think hidden menu items make a lot sense to me as presented.

I guess here I'll put that hacking vendors is a engineering field, and from an Engineer's perspective the vending machines seem to be very consistent in a few ways

They take identical restock kits- I always thought compressed matter

They seem to usually product product on demand
- Monkey vendors
- When you destroy them, they usually take their inventory with them
- Only a 25% chance they enter a "malfunction" state and actually produce inventory on destruction

So investigation by an engineer reveals

Banking functions are identical
The service panels are identical
It malfunctions identically, generating and throwing inventory when the appropriate wire is puled
Their PDA interfaces with it just like a vending machine

So while there's some suspension of disbelief, it's much lesser to any engineer or anyone with tools- the people who hack vending machines exactly.

**Comment by penny64** [comment]
I'll try to get this merge conflict fixed soon thanks for the approve

**Pull request merged**